File: python/Program/Lampu.py
from PerabotanRumahTangga import PerabotanRumahTangga

# =============================
# Class Lampu
# Level 3 (Child dari PerabotanRumahTangga)
# =============================
class Lampu(PerabotanRumahTangga):
    def __init__(self, nama_produk="", merek="", harga=0,
                 kategori="", material="", berat=0.0,
                 daya=0, jenis_lampu="", garansi=0):
        """
        Inisialisasi atribut lampu
        Memanggil constructor parent
        """
        super().__init__(nama_produk, merek, harga, kategori, material, berat)
        self.daya = daya
        self.jenis_lampu = jenis_lampu
        self.garansi = garansi

    # Getter dan Setter
    def get_daya(self):
        return self.daya

    def set_daya(self, daya):
        self.daya = daya

    def get_jenis_lampu(self):
        return self.jenis_lampu

    def set_jenis_lampu(self, jenis_lampu):
        self.jenis_lampu = jenis_lampu

    def get_garansi(self):
        return self.garansi

    def set_garansi(self, garansi):
        self.garansi = garansi

File: java/Program/Lampu.java
// =============================
// Class Lampu
// Level 3 (Child dari PerabotanRumahTangga)
// =============================

public class Lampu extends PerabotanRumahTangga {
    // =============================
    // Atribut khusus untuk lampu
    // =============================
    private int daya;         // daya listrik lampu dalam watt
    private String jenisLampu;// jenis lampu (LED, Fluorescent, Halogen, dll)
    private int garansi;      // masa garansi lampu dalam bulan

    // =============================
    // Constructor kosong (default)
    // =============================
    public Lampu() {}

    // =============================
    // Constructor dengan parameter
    // Memanggil parent constructor dan inisialisasi semua atribut
    // =============================
    public Lampu(String namaProduk, String merek, int harga,
                 String kategori, String material, float berat,
                 int daya, String jenisLampu, int garansi) {
        super(namaProduk, merek, harga, kategori, material, berat); // panggil constructor parent
        this.daya = daya;
        this.jenisLampu = jenisLampu;
        this.garansi = garansi;
    }

    // =============================
    // Getter dan Setter
    // =============================
    public int getDaya() { return daya; }
    public String getJenisLampu() { return jenisLampu; }
    public int getGaransi() { return garansi; }

    public void setDaya(int daya) { this.daya = daya; }
    public void setJenisLampu(String jenisLampu) { this.jenisLampu = jenisLampu; }
    public void setGaransi(int garansi) { this.garansi = garansi; }
}